### PR TITLE
chore: Automates changing Terraform supported versions in provider documentation

### PR DESCRIPTION
**Pull request by maastha**
## Description

Automate changing Terraform supported versions on documentation. This PR:
- adds script to update website/docs/index.html.markdown with supported TF version information
- adds github action to run the job daily and trigger a PR if any changes

Link to any related issue(s): [CLOUDP-237003](https://jira.mongodb.org/browse/CLOUDP-237003)

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue). Please, add the "bug" label to the PR.
- [x] New feature (non-breaking change which adds functionality). Please, add the "enhancement" label to the PR.
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected). Please, add the "breaking change" label to the PR.
- [ ] This change requires a documentation update
- [ ] Documentation fix/enhancement

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have read the [contribution guidelines](https://github.com/mongodb/terraform-provider-mongodbatlas/blob/master/CONTRIBUTING.md)
- [x] I have checked that this change does not generate any credentials and that **they are NOT accidentally logged anywhere**.
- [x] I have added tests that prove my fix is effective or that my feature works per HashiCorp requirements
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run make fmt and formatted my code
- [x] If changes include deprecations or removals, I defined an isolated PR with a relevant title as it will be used in the auto-generated changelog.
- [x] If changes include removal or addition of 3rd party GitHub actions, I updated our internal document. Reach out to the APIx Integration slack channel to get access to the internal document.

## Testing
Test PR created from workflow on forked repository: https://github.com/maastha/terraform-provider-mongodbatlas-fork/pull/1 

## Further comments
Modified existing script to take an argument to output different types of supported version information:
```
scripts git:(CLOUDP-237003-automate-matrix-doc) ✗ ./get-terraform-supported-versions.sh true
[
  {
    "version": "v1.7.0",
    "published_at": "2024-01-17T19:59:32Z",
    "end_support_date": "2026-01-31T19:59:32Z"
  },
  {
    "version": "v1.6.0",
    "published_at": "2023-10-04T17:11:35Z",
    "end_support_date": "2025-10-31T17:11:35Z"
  },
  {
    "version": "v1.5.0",
    "published_at": "2023-06-12T10:13:09Z",
    "end_support_date": "2025-06-30T10:13:09Z"
  },
  {
    "version": "v1.4.0",
    "published_at": "2023-03-08T18:13:32Z",
    "end_support_date": "2025-03-31T18:13:32Z"
  },
  {
    "version": "v1.3.0",
    "published_at": "2022-09-21T13:59:25Z",
    "end_support_date": "2024-09-30T13:59:25Z"
  },
  {
    "version": "v1.2.0",
    "published_at": "2022-05-18T21:47:47Z",
    "end_support_date": "2024-05-31T21:47:47Z"
  }
]


➜  scripts git:(CLOUDP-237003-automate-matrix-doc) ✗ ./get-terraform-supported-versions.sh false
["1.7.x","1.6.x","1.5.x","1.4.x","1.3.x","1.2.x"]

```

### CODE REVIEW

File: scripts/get-terraform-supported-versions.sh [hunk 1]
```diff
@@ -14,26 +14,35 @@
...
-		02)
-			# February: check if it's a leap year
-			if (( year % 4 == 0 && ( year % 100 != 0 || year % 400 == 0 ) )); then
-				last_day_of_month=29
-			else
-				last_day_of_month=28
-			fi
-			;;
-		esac
+	01 | 03 | 05 | 07 | 08 | 10 | 12)
```
Comment (maastha) [inline scripts/get-terraform-supported-versions.sh:28]: formatted

**Comment by oarbusi** [comment]
I would change the PR title to `chore` since this is an internal improvement not a customer facing feature

**Review by marcosuma — APPROVED** [review]

**Review by oarbusi — APPROVED** [review]

**Review by AgustinBettati — APPROVED** [review]

File: website/docs/index.html.markdown [hunk 2]
```diff
@@ -189,7 +191,7 @@ For more information on configuring and managing programmatic API Keys see the [
 | 1.4.x | 2023-03-08 | 2025-03-31 | 2025-03-31 |
 | 1.3.x | 2022-09-21 | 2024-09-30 | 2024-09-30 |
 | 1.2.x | 2022-05-18 | 2024-05-31 | 2024-05-31 |
-
+<!-- MATRIX_PLACEHOLDER_END -->
```
Comment (lantoli) [inline website/docs/index.html.markdown:194]: a nice test would be to have this empty in the PR and see how tomorrow the PR is generated to create the table

Comment (maastha) [reply website/docs/index.html.markdown:194]: I tried running the job manually in the fork and PR was created https://github.com/maastha/terraform-provider-mongodbatlas-fork/pull/1
WDYT?

File: .github/workflows/update_tf_compatibility_matrix.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+name: Update Terraform Compatibility Matrix documentation
```
Comment (lantoli) [inline .github/workflows/update_tf_compatibility_matrix.yml:1]: as we're already in the TF repo, I would remove it from the file and title

File: scripts/get-terraform-supported-versions.sh [hunk 4]
```diff
@@ -14,26 +14,35 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 set -euo pipefail
+
```
Comment (lantoli) [inline scripts/get-terraform-supported-versions.sh:17]: same, I would remove TF from filename

File: scripts/update-tf-compatibility-matrix.sh [hunk 5]
```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/env bash
```
Comment (EspenAlbert) [inline scripts/update-tf-compatibility-matrix.sh:1]: Another big shell script 😅 Did you consider using python?

Comment (lantoli) [reply scripts/update-tf-compatibility-matrix.sh:1]: or Go as Agustin did for the one in CFN listing versions in regions ;-)

a great topic you may want to bring to next tech meeting

File: .github/workflows/update_tf_compatibility_matrix.yml (hunk 3 above)
Comment (oarbusi) [reply .github/workflows/update_tf_compatibility_matrix.yml:1]: I don't fully agree, just using "compatibility matrix" could refer to terraform, terraform plugin framework, terraform provider, etc and cause confusion.

File: scripts/update-tf-compatibility-matrix.sh (hunk 5 above)
Comment (maastha) [reply scripts/update-tf-compatibility-matrix.sh:1]: yeah I essentially re-used the existing script

Comment (lantoli) [reply scripts/update-tf-compatibility-matrix.sh:1]: or Go as @AgustinBettati  did for the one in CFN listing CFN versions in regions ;-) 
Go templates are nice.

a great topic you may want to bring to next tech meeting

File: .github/workflows/update_tf_compatibility_matrix.yml (hunk 3 above)
Comment (maastha) [reply .github/workflows/update_tf_compatibility_matrix.yml:1]: I agree with @oarbusi I don't think it's redundant

**Pull request merged**